public/app/features/alerting/unified/components/receivers/ReceiversTable.tsx

```diff
@@ -1,23 +1,64 @@
-import { useStyles2 } from '@grafana/ui';
+import { Button, ConfirmModal, Modal, useStyles2 } from '@grafana/ui';
 import { AlertManagerCortexConfig } from 'app/plugins/datasource/alertmanager/types';
-import React, { FC, useMemo } from 'react';
+import React, { FC, useMemo, useState } from 'react';
 import { useUnifiedAlertingSelector } from '../../hooks/useUnifiedAlertingSelector';
 import { getAlertTableStyles } from '../../styles/table';
 import { extractReadableNotifierTypes } from '../../utils/receivers';
 import { ActionIcon } from '../rules/ActionIcon';
 import { ReceiversSection } from './ReceiversSection';
 import { makeAMLink } from '../../utils/misc';
+import { isReceiverUsed } from '../../utils/alertmanager-config';
+import { useDispatch } from 'react-redux';
+import { updateAlertManagerConfigAction } from '../../state/actions';
 
 interface Props {
   config: AlertManagerCortexConfig;
   alertManagerName: string;
 }
 
 export const ReceiversTable: FC<Props> = ({ config, alertManagerName }) => {
+  const dispatch = useDispatch();
   const tableStyles = useStyles2(getAlertTableStyles);
 
   const grafanaNotifiers = useUnifiedAlertingSelector((state) => state.grafanaNotifiers);
 
+  // receiver name slated for deletion. If this is set, a confirmation modal is shown. If user approves, this receiver is deleted
+  const [receiverToDelete, setReceiverToDelete] = useState<string>();
+  const [showCannotDeleteReceiverModal, setShowCannotDeleteReceiverModal] = useState(false);
+
+  const onClickDeleteReceiver = (receiverName: string): void => {
+    if (isReceiverUsed(receiverName, config)) {
+      setShowCannotDeleteReceiverModal(true);
+    } else {
+      setReceiverToDelete(receiverName);
+    }
+  };
+
+  const deleteReceiver = () => {
+    if (
```

Comment:
What do you think about moving this if-statement into an action instead? So basically always dispatch a thunk action that does this check in it.

Would make the code easier to test and also clean up the component a bit.

Reply:
makes sense, done! the same for templates as well